File: Linked-lists/singly/insertAtEnd.cpp
#include <bits/stdc++.h>
using namespace std;

struct NODE
{
    int data;
    NODE *next;
};

NODE *head = nullptr;

NODE *create(int value)
{
    NODE *new_node = new NODE;

    new_node->data = value;
    new_node->next = nullptr;

    if (head == nullptr)
    {
        head = new_node;
    }
    else
    {
        NODE *temp = head;
        while (temp->next != nullptr)
        {
            temp = temp->next;
        }
        temp->next = new_node;
    }

    return new_node;
}

NODE *insertAtEnd(int value)
{
    NODE *new_node = new NODE;
    new_node->data = value;
    new_node->next = nullptr;

    if (head == nullptr)
    {
        head = new_node;
    }
    else
    {
        NODE *temp = head;
        while (temp->next != nullptr)
        {
            temp = temp->next;
        }
        temp->next = new_node;
    }

    return new_node;
}

void print(NODE *node)
{
    while (node != nullptr)
    {
        cout << node->data << " ";
        node = node->next;
    }
    cout << "null" << endl;
}

int main()
{
    create(1);
    create(2);
    create(3);
    create(4);

    print(head);

    insertAtEnd(11);

    print(head);

    return 0;
}

File: Linked-lists/singly/insertAtEnd.java
public class insertAtEnd {

    static class Node {
        int data;
        Node next;

        Node(int value) {
            data = value;
            next = null;
        }
    }

    // Head of the linked list
    static Node head = null;

    // Method to create a new node and add it to the linked list
    public static Node create(int value) {
        Node newNode = new Node(value);

        if (head == null) {
            head = newNode;
        } else {
            Node temp = head;
            while (temp.next != null) {
                temp = temp.next;
            }
            temp.next = newNode;
        }

        return newNode;
    }

    // Method to insert a new node at the end of the list
    public static Node InsertAtEnd(int value) {
        Node newNode = new Node(value);

        if (head == null) {
            head = newNode;
        } else {
            Node temp = head;
            while (temp.next != null) {
                temp = temp.next;
            }
            temp.next = newNode;
        }

        return newNode;
    }

    // Method to print the linked list
    public static void print(Node node) {
        while (node != null) {
            System.out.print(node.data + " ");
            node = node.next;
        }
        System.out.println("null");
    }

    public static void main(String[] args) {
        // Create nodes with values
        create(1);
        create(2);
        create(3);
        create(4);

        // Print the linked list
        print(head);

        // Insert node at the end
        InsertAtEnd(11);

        // Print the linked list again
        print(head);
    }
}
